_studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_ddi.cpp

```diff
@@ -40,9 +40,13 @@ GUID GetGUID(MfxVideoParam const & par)
 #if (MFX_VERSION >= 1027)
     if (par.mfx.CodecProfile == MFX_PROFILE_HEVC_MAIN10 || par.m_ext.CO3.TargetBitDepthLuma == 10)
         bdId = 1;
+    if (par.m_ext.CO3.TargetBitDepthLuma == 12)
+        bdId = 2;
 
     cfId = mfx::clamp<mfxU16>(par.m_ext.CO3.TargetChromaFormatPlus1 - 1, MFX_CHROMAFORMAT_YUV420, MFX_CHROMAFORMAT_YUV444) - MFX_CHROMAFORMAT_YUV420;
 
+    if (par.m_platform && par.m_platform < MFX_HW_ICL)
```

Comment:
Which profiles are supported and which are not is information which can be _queried_ from the driver. Why there is any hardcoded platform check here?

Reply:
I'd like to involve @saosipov into this discussion.